### PR TITLE
Style Menu Options

### PR DESCRIPTION
**Pull request by 0x747972**
Please review the code.
Thanks.

Closes: #1181

### CODE REVIEW

**Comment by flatcap** [comment]
I've added a commit to eliminate `pos`.
Left to do:
- Check for closing `)`
  Code is confused by `he(ll)o`
- Eliminate `1: Apple` style questions
  - `ncrypt/smime.c`
  - `po/*.po`

**Comment by gahr** [comment]
I think 2b70591 is over-complicated. I understand the need to check for '(x)', but going through the string three times isn't probably the best way to do it.

How about:

```c
 while ((cur = strchr(prompt, '(')))
{
  // write the part between prompt and cur using MT_COLOR_PROMPT
  SETCOLOR(MT_COLOR_PROMPT);
  addnstr(prompt, cur - prompt);

  if (cur[1] && cur[2] == ')')
  {
    // we have a single letter within parentheses
    SETCOLOR(MT_COLOR_OPTIONS);
    addnstr(cur + 1, 1);
    prompt = cur + 3;
  }
  else
  {
    // we have a parenthesis followed by something else
    SETCOLOR(MT_COLOR_PROMPT);
    addnstr(cur, 1);
    prompt += cur + 1;
  }
}

File: curs_lib.c [hunk 1]
```diff
@@ -821,9 +820,40 @@ int mutt_multi_choice(const char *prompt, const char *letters)
...
       }
 
+      mutt_window_move(MuttMessageWindow, 0, 0);
+
+      if ((ColorDefs[MT_COLOR_OPTIONS] != 0) &&
+          (ColorDefs[MT_COLOR_OPTIONS] != ColorDefs[MT_COLOR_PROMPT]))
+      {
+        char *cur = NULL;
+
+        while ((cur = strchr(prompt, '(')))
```
[the remark this answers is not in the document]
Comment (flatcap) [reply curs_lib.c]: Don't worry about this.
I dislike assignment-test conditions, but some cases, like this `while` loop need it, or risk duplication of the assignment.

File: curs_lib.c [hunk 2]
```diff
@@ -821,9 +820,40 @@ int mutt_multi_choice(const char *prompt, const char *letters)
...
+          {
+            // we have a single letter within parentheses
+            SETCOLOR(MT_COLOR_OPTIONS);
+            addnstr(cur + 1, 1);
+            prompt = cur + 3;
+          }
+          else
+          {
+            // we have a parenthesis followed by something else
+            SETCOLOR(MT_COLOR_PROMPT);
```
[the remark this answers is not in the document]
Comment (gahr) [reply curs_lib.c]: Definitely, thanks!

File: curs_lib.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (gahr) [reply curs_lib.c]: I'm glad to see that the entry is not part of the strict *rules*, but rather of the somewhat more lax *tips* :) I think it makes sense for ifs, where you execute the expression once anyway. For whiles, where you would end up executing the same expression before and at the end of the body, it makes zero sense :) same goes for the for-loop, where you have to repeat the expression in the precondition and in the increment part.

[the remark this answers is not in the document]
Comment (gahr) [reply curs_lib.c]: @0x747972 can you please undo this part?

**Comment by flatcap** [comment]
I've transplanted two other commits to the 'translate' branch:
- update German translation
- smime: unify multi-option questions

I'll merge them shortly.

**Pull request merged**